app/controllers/downloads_controller.rb

```diff
@@ -1,217 +1,51 @@
 class DownloadsController < ApplicationController
 
-  include XhtmlSplitter
-
   skip_before_action :store_location, only: :show
+  before_action :load_work, only: :show
   before_action :check_visibility, only: :show
+  after_action :remove_downloads, only: :show
 
-  # named route: download_path
-  # Note: only :id and :format need to be correct,
-  # the other two are derived and are there for nginx's benefit
-  # GET /downloads/:download_prefix/:download_authors/:id/:download_title.:format
   def show
-    @work = Work.find(params[:id])
-    @check_visibility_of = @work
+    respond_to :html, :pdf, :mobi, :epub
+    @download = Download.generate(@work, mime_type: request.format)
 
-    if @work.unrevealed?
-      flash[:error] = ts("Sorry, you can't download an unrevealed work")
-      redirect_back_or_default works_path
+    # Make sure we were able to generate the download
+    unless @download.present? && @download.exists?
+      flash[:error] = ts("We were not able to render this work. Please try again in a little while or try another format.")
+      redirect_to work_path(@work)
       return
     end
 
-    unless @admin_settings.downloads_enabled?
```

Comment:
Do we want to remove this admin setting?

Reply:
I think this was probably a victim of a merge; I'm going to put it back.